zng/ztests/enum-err.yaml

```diff
@@ -0,0 +1,9 @@
+zql: "put s=e:string"
+
+input: |
+  #0:record[e:enum[int32,foo:[1],bar:[2],baz:[4]]]
+  0:[100;]
```

Comment:
shouldn't this just be invalid tzng/zng rather than an error that is discovered when you try to use it?

Reply:
We don't have a semantic checker yet in the zq system but yes, if the index is a constant (it doesn't have to be), then we should check for this at compile time.  e.g., this current works...
```
"head 1 | put x=e[e]" enum.tzng

Reply:
Hmm, I was just saying that 
```
#0:record[e:enum[int32,foo:[1],bar:[2],baz:[4]]]
  0:[100;]
```
could be determined as invalid zng when reading the record, similar to how `Record.Typecheck()` catches some invalid encodings.  Maybe I'm misunderstanding something here?

Reply:
Sorry, you're right.  Will address.